openpype/pipeline/publish/lib.py

```diff
@@ -623,12 +641,58 @@ def get_instance_staging_dir(instance):
     Returns:
         str: Path to staging dir of instance.
     """
+    staging_dir = instance.data.get('stagingDir')
+    if staging_dir:
+        return staging_dir
+
+    openpype_temp_dir = os.getenv("OPENPYPE_TEMP_DIR")
+    custom_temp_dir = None
+    if openpype_temp_dir:
+        if "{" in openpype_temp_dir:
+            custom_temp_dir = _formated_staging_dir(
+                instance, openpype_temp_dir
+            )
+        elif os.path.exists(openpype_temp_dir):
```

Comment:
So if 'OPENPYPE_TEMP_DIR' is set but the directory does not exist (e.g. "F:/MyPath/temp") if won't be used?

Reply:
That's seems odd, especially with a potentially formatted staging dir path I'd assume at least an `os.makedirs` would have been tried so that it could be dynamic folders?

Reply:
@iLLiCiTiT you are right, this use case can happen, but in case there is template `_format_staging_dir` is having its own `os.makedirs`

Reply:
Ok, but then if it _were_ a path without template variables but a hardcoded path (e.g. a hardcoded local temp dir) it wouldn't get generated? Seems odd maybe?

Reply:
If I would (as artist) set environment `OPENPYPE_TMPDIR` to `F:/MyPath/temp` but the folder does not exists and because it does not contain `{` it won't get to `_format_staging_dir` and thus won't be used.

Reply:
But would you expect that, or not? I would be confused as an artist. :)